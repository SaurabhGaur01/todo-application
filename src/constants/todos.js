export const ADD_TODO = 'modules/todo-app/ADD_TODO';
export const CLEAR_TODO = 'modules/todo-app/CLEAR_TODO';
export const MASS_ADD = 'modules/todo-app/MASS_ADD';
export const TOGGLE_TODO = 'modules/todo-app/TOGGLE_TODO';