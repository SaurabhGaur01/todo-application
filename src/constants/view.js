export const VIEW_SHOW_ALL = 'SHOW_ALL';
export const VIEW_SHOW_COMPLETED = 'SHOW_COMPLETED';
export const VIEW_SHOW_ACTIVE = 'SHOW_ACTIVE';