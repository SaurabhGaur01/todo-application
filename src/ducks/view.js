import { VIEW_SHOW_ALL } from '../constants/view';

export const SET_VIEW = 'modules/todo-app/SET_VIEW';

export const setView = (filter) => ({
    type: SET_VIEW,
    filter
})

export const viewReducer = (state = VIEW_SHOW_ALL, action = { type: 'NULL_ACTION'}) => {
    switch (action.type) {
      case SET_VIEW:
        return action.filter
      default:
        return state
    }
}