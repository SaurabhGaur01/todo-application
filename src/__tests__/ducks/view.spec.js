import {
    setView,
    viewReducer,
} from '../../ducks/view';

import {  VIEW_SHOW_ALL } from '../../constants/view';

describe('view Duck Tests', () => {
    const mockState = {
        key: 'mockKey',
    }

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setView should match the snapshot' , () => {
            expect(setView()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(viewReducer()).toEqual(VIEW_SHOW_ALL);
        });
        it('should return the passed state if called with no action' , () => {
            expect(viewReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(viewReducer(mockState, unknownAction)).toBe(mockState);
        });
    });
})