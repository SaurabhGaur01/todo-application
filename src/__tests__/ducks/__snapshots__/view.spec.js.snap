// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view Duck Tests Action Generator setView should match the snapshot 1`] = `
Object {
  "filter": undefined,
  "type": "modules/todo-app/SET_VIEW",
}
`;
